### PR TITLE
[ticket/14688] Add core events to the feeds

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

<a href="https://tracker.phpbb.com/browse/PHPBB3-14688">PHPBB3-14688</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Event

File: phpBB/phpbb/feed/base.php [hunk 1]
```diff
@@ -81,8 +84,18 @@
...
 	* @param string								$phpEx		php file extension
+	* @param \phpbb\event\dispatcher_interface	$phpbb_dispatcher		Event dispatcher object
 	*/
-	function __construct(\phpbb\feed\helper $helper, \phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\cache\driver\driver_interface $cache, \phpbb\user $user, \phpbb\auth\auth $auth, \phpbb\content_visibility $content_visibility, $phpEx)
+	function __construct(
+		\phpbb\feed\helper $helper,
+		\phpbb\config\config $config,
+		\phpbb\db\driver\driver_interface $db,
+		\phpbb\cache\driver\driver_interface $cache,
+		\phpbb\user $user, \phpbb\auth\auth $auth,
```
Comment (lavigor) [inline phpBB/phpbb/feed/base.php]: Probably each parameter on a separate line will be better?

Comment (rxu) [inline phpBB/phpbb/feed/base.php]: Sure, just overlooked while editing, thanks.

File: phpBB/phpbb/feed/base.php [hunk 2]
```diff
@@ -81,8 +84,19 @@
 	* @param \phpbb\auth\auth					$auth		Auth object
 	* @param \phpbb\content_visibility			$content_visibility		Auth object
 	* @param string								$phpEx		php file extension
+	* @param \phpbb\event\dispatcher_interface	$phpbb_dispatcher		Event dispatcher object
 	*/
-	function __construct(\phpbb\feed\helper $helper, \phpbb\config\config $config, \phpbb\db\driver\driver_interface $db, \phpbb\cache\driver\driver_interface $cache, \phpbb\user $user, \phpbb\auth\auth $auth, \phpbb\content_visibility $content_visibility, $phpEx)
+	function __construct(
```
Comment (marc1706) [inline phpBB/phpbb/feed/base.php]: In general, constructor arguments should be classes first and then strings and optional parts at the end. Please move the dispatcher in front of the `$phpEx`

**Comment by rxu** [comment]
Updated #4365.

**Pull request merged**